kornia/geometry/fundamental.py

```diff
@@ -0,0 +1,138 @@
+"""Module containing operators to work with fundamental and essential matrices."""
+
+import torch
+from kornia.geometry.conversions import convert_points_to_homogeneous
+
+
+def raise_error_if_pts_are_not_valid(pts: torch.Tensor, tensor_name: str = 'pts') -> None:
+    """Auxilary function, which verifies that input is a torch.tensor of [BxNx2] or [BxNx3] shape
+
+    Args:
+        pts (any): input to test
+        tensor_name (str): tensor name for error message
+    """
+    if not isinstance(pts, torch.Tensor):
+        raise TypeError("{} type is not a torch.Tensor. Got {}".format(
+            tensor_name, type(pts)))
+
+    if (len(pts.shape) != 3) or (pts.size(-1) not in [2, 3]):
+        raise ValueError(
+            "{} must be a (B, N, 3) or (B, N, 2) tensor. Got {}".format(
+                tensor_name, pts.shape))
+    return
+
+
+def symmetrical_epipolar_distance(pts1: torch.Tensor,
+                                  pts2: torch.Tensor,
+                                  Fm: torch.Tensor,
+                                  squared: bool = True,
+                                  eps: float = 1e-9) -> torch.Tensor:
```

Comment:
I'm gonna push all the sfm stuff in a couple of days. Are you finishing this one ? otherwise I can include it in my PR. This is under `kornia.geometry.fundamental` now. What about merging everything related to fundamental, essential, triangulation to `kornia.sfm`. The homography related functionalities could go maybe to `kornia.geometry.perspective` ?

Reply:
Feel free to copy-past. I am working on it, but in kind of slower pace. Regarding naming - agree

Reply:
great :)